### ISSUE TITLE
Improve the mechanism for skipping elements of parameter arrays with connection rule all_to_all

### ISSUE THREAD

**Issue by golosio** [nest/nest-simulator#519]
in conn_builder.cpp, line 849-853:

```
  for ( GIDCollection::const_iterator sgid = sources_->begin();
        sgid != sources_->end();
        ++sgid )
    skip_conn_parameter_( tid );
```

and same thing in other parts of the same file.
Looping over skip_conn_parameter_() can become very inefficient due to nested loops and function calls, including to ConnParameter::skip(), which requires a vtable lookup. Since the number of source neurons is fixed in the all_to_all rule, it should be possible to avoid those loops and to pass the number of source neurons as second argument of the call to skip_conn_parameter. This can be done once PR https://github.com/nest/nest-simulator/pull/465 is merged.

**Comment by heplesser** [comment]
@golosio Note that #465 is merged and that a version of `skip_conn_parameter_()` skipping a given number of nodes is now available.

**Comment by golosio** [comment]
@heplesser right, I can take care of this issue

**Comment by heplesser** [comment]
Fixed by #568.
